src/code_transformation/differentiable.jl

```diff
@@ -95,13 +95,13 @@ function unionise_struct(code::Expr)
         curly = Expr(:curly, name.args[1], unionise_subtype.(name.args[2:end])...)
         if is_subtype_expr
             return Expr(
-                :type,
+                :struct,
```

Comment:
Does mutability matter here?

Reply:
`:struct` is the expression head for both mutable and immutable types; the `Bool` argument dictates whether it's immutable. This change shouldn't affect behavior at all.

Reply:
Excellent. I thought maybe that was the case.